src/services/timeout.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2018-2020 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+export const timeout = (ms = 1000) => {
```

Comment:
Name `timeout` you've chosen seems very misleading to me. This service is used to add a delay into workspace loading (or factory loading) flow, but not as a timeout. Please consider to rename it.

Reply:
It was called 'timeout' because of using 'setTimeout' inside.  I agree to rename it:
```timeout``` -> ```delay```